openquake/hazardlib/tests/source/simple_fault_test.py

```diff
@@ -648,3 +648,78 @@ def test_slip_vertical_rupture(self):
             self.assertAlmostEqual(rup.rupture_slip_direction,
                                    slip[i], delta=0.1)
             self.assertAlmostEqual(rup.occurrence_rate, rate[i], delta=0.01)
+
+
+class ModifySimpleFaultTestCase(_BaseFaultSourceTestCase):
+    """
+    Tests all of the geometry modification methods
+    """
+    def setUp(self):
+        self.basic_trace = Line([Point(30.0, 30.0), Point(30.0, 32.0)])
+        self.mfd = EvenlyDiscretizedMFD(7.0, 0.1, [1.0])
+        self.aspect = 1.0
+        self.dip = 45.0
+        self.fault = self._make_source(self.mfd, self.aspect, self.basic_trace,
+                                       self.dip)
+        self.fault.lower_seismogenic_depth = 10.0
+
+    def test_modify_set_geometry_trace(self):
+        new_trace = Line([Point(30.0, 30.0), Point(30.2, 32.25)])
+        self.fault.modify_set_geometry(new_trace, 0., 10., 45.)
```

Comment:
Instead of modifying the original fault and then restoring the initial value, don't you define a 

`fault = copy.deepcopy(self.fault)`

and modify the copy?